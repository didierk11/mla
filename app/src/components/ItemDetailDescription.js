import React from "react";

const getItemDetails = (props) => {
  const descriptionText = props.description;
  return (
    <div className="col-lg-7 item-detail-description">
      <p className="item-detail-description-title">Descripción del Producto</p>
      <p className="item-detail-description-text">
        {descriptionText
          ? descriptionText
          : "El vendedor no proporcionó una descripción."}
      </p>
    </div>
  );
};

export default getItemDetails;

