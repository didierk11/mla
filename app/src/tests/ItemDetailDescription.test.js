import React from "react";
import { render } from "react-dom";
import ItemDetailDescription from "../components/ItemDetailDescription";
import renderer from "react-test-renderer";

//testing data
const desc = 'Item valid description text.';

describe("Snapshot test:", () => {
  it("ItemDetailDescription render correctly", () => {
    const div = renderer.create(<ItemDetailDescription description={desc} />).toJSON();
    expect(div).toMatchSnapshot();
  });
});

describe("Smoke test:", () => {
  it("ItemDetailDescription render without crashing", () => {
    const div = document.createElement("div");
    render(<ItemDetailDescription description={desc} />, div);
  });
});