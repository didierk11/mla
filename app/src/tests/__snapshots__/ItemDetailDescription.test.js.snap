// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test: ItemDetailDescription render correctly 1`] = `
<div
  className="col-lg-7 item-detail-description"
>
  <p
    className="item-detail-description-title"
  >
    Descripci√≥n del Producto
  </p>
  <p
    className="item-detail-description-text"
  >
    Item valid description text.
  </p>
</div>
`;
